app/src/main/java/com/example/mystore/viewmodels/ProductViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.mystore.viewmodels
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.example.mystore.ProductsRepository
+import com.example.mystore.models.ProductModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class ProductDetailsViewModel @Inject constructor(
+    private val repository: ProductsRepository
+) : ViewModel() {
+    var name by mutableStateOf("")
```

Comment:
What I intended with my previous comment was to not rely on those `mutableStateOf`s here at all.

You could simulate a network request like this:

```
@HiltViewModel
class ProductDetailsViewModel @Inject constructor(
    private val repository: ProductsRepository
) : ViewModel() {
    var product: ProductModel? by mutableStateOf(null)

    init {
        fetchProduct()
    }

    private fun fetchProduct() {
        product = ProductModel(
            name = "Stargizer's Tea Set",
            description = "Include in our heavenly tea experience with our Stargizer's Tea Set, featuring a constellation-themed teapot and two matching teacups. Crafted from fine porcelain, this elegant set will transport you to the cosmos with every sip.",
            category = "Home",
            price = 90.00f,
            rating = 5
        )
    }
}
```

And then use this public optional `product` state in the composable with a null-check operator, i.e.:

`Text(text = viewModel.product?.name ?: "")`

I'm leaving this comment but approving the PR so you could go ahead and merge it if you wish and address this in the next homework's PR where you'll have some actual networking implementation. You've already done great with the first task and that one is slightly out of scope for it anyway.